### PR TITLE
Refactor FXIOS-3966 [v104] Remove background sync logic from app delegate

### PR DESCRIPTION
**Pull request by OrlaM**
Re-opening to try to fix the UI test issue.

### CODE REVIEW

**Comment by isabelrios** [comment]
> It seems the order of operations in applicationDidEnterBackground is important in ways I'm having a hard time reasoning about so as a work around for now I'll just remove the notification from the utility and call it directly from applicationDidEnterBackground. cc @isabelrios

yay I see all tests passed now! yeah, for some reason that order was breaking the login view... happy you found a workaround, thanks!

File: Client/Application/BackgroundSyncUtil.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import BackgroundTasks
+
+class BackgroundSyncUtil {
+
+    let profile: Profile
+    let application: UIApplication
```
Comment (electricRGB) [inline Client/Application/BackgroundSyncUtil.swift:11]: I don't know why I'm confused and I feel silly asking but...... should this be passed in as a class property? Does it matter if this class is hanging on to a ref to the application? I've never actually dealt with passing the UIApplication itself around so I just don't know.

File: Client/Application/BackgroundSyncUtil.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        guard application.applicationState != .active else { return }
+
+        profile._shutdown()
+    }
+
+    // MARK: Notifiable
+
+    func handleNotifications(_ notification: Notification) {
+        switch notification.name {
+        case UIApplication.willResignActiveNotification:
```
Comment (electricRGB) [inline Client/Application/BackgroundSyncUtil.swift:91]: I've probably missed it, but where are we watching for this?

Comment (OrlaM) [reply Client/Application/BackgroundSyncUtil.swift:91]: Missed it in clean up, will remove.

**Review by electricRGB — APPROVED** [review]

**Pull request merged**